controllers/neutronapi_controller.go

```diff
@@ -809,6 +832,71 @@ func (r *NeutronAPIReconciler) transportURLCreateOrUpdate(instance *neutronv1bet
 	return transportURL, op, err
 }
 
+// TODO(ihar) this function could live in lib-common
+// deleteExternalSecrets - delete obsolete secrets for external dataplane consumption
+func (r *NeutronAPIReconciler) deleteExternalSecrets(
+	ctx context.Context,
+	h *helper.Helper,
+	instance *neutronv1beta1.NeutronAPI,
+) error {
+	cm := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-external-config", instance.Name),
+			Namespace: instance.Namespace,
+		},
+	}
+
+	err := h.GetClient().Delete(ctx, cm)
+	if err != nil && !k8s_errors.IsNotFound(err) {
+		return fmt.Errorf("error deleting external secret %s: %w", cm.Name, err)
+	}
+	return nil
+}
+
+// generateExternalSecrets - create secrets for external dataplane consumption
+func (r *NeutronAPIReconciler) generateExternalSecrets(
+	ctx context.Context,
+	h *helper.Helper,
+	instance *neutronv1beta1.NeutronAPI,
+	sbCluster *ovnclient.OVNDBCluster,
+	envVars *map[string]env.Setter,
+) error {
+	// Create/update secrets from templates
+	cmLabels := labels.GetLabels(instance, labels.GetGroupLabel(neutronapi.ServiceName), map[string]string{})
+
+	templates := map[string]string{
+		"neutron_ovn_metadata_agent.ini": "/neutron_ovn_metadata_agent.ini",
+	}
+
+	transportURLSecret, _, err := secret.GetSecret(ctx, h, instance.Status.TransportURLSecret, instance.Namespace)
+	if err != nil {
+		return err
+	}
+
+	sbEndpoint, err := sbCluster.GetExternalEndpoint()
+	if err != nil {
+		return err
+	}
+
+	templateParameters := make(map[string]interface{})
+	templateParameters["TransportURL"] = string(transportURLSecret.Data["transport_url"])
+	templateParameters["SBConnection"] = sbEndpoint
+
+	cms := []util.Template{
+		// external Secret
+		{
+			Name:               fmt.Sprintf("%s-external-config", instance.Name),
+			Namespace:          instance.Namespace,
+			Type:               util.TemplateTypeNone,
+			InstanceType:       instance.Kind,
+			Labels:             cmLabels,
+			ConfigOptions:      templateParameters,
+			AdditionalTemplate: templates,
+		},
+	}
```

Comment:
Nova also records the hash of the Secret in its Status to serve as a reference point for comparison with what was used for the deployment. See https://github.com/openstack-k8s-operators/nova-operator/pull/515/files#diff-86a1ea2f9281135a424412db58e53334e5e0e4dbf4e2f6b0a315daae0f470999R591